### PR TITLE
[SPARK-30458][WEBUI] Fix Wrong Executor Computing Time in Time Line of Stage Page

### PR DESCRIPTION
**Pull request by sddyljsx**
### What changes were proposed in this pull request?
The Executor Computing Time in Time Line of Stage Page will be right

### Why are the changes needed?
The Executor Computing Time in Time Line of Stage Page is Wrong. It includes the Scheduler Delay Time, while the Proportion excludes the Scheduler Delay

Snipaste_2020-01-08_19-04-33

The right executor computing time is 1ms, but the number in UI is 3ms(include 2ms scheduler delay); the proportion is right.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Manual

### CODE REVIEW

**Comment by sddyljsx** [comment]
Snipaste_2020-01-08_19-04-33

**Comment by lw-lin** [comment]
Jenkins test this please

**Comment by sddyljsx** [comment]
@HyukjinKwon PTAL

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by HyukjinKwon** [comment]
cc @gengliangwang

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala [hunk 1]
```diff
@@ -288,10 +288,10 @@ private[ui] class StagePage(parent: StagesTab, store: AppStatusStore) extends We
 
         val executorOverhead = serializationTime + deserializationTime
         val executorRunTime = if (taskInfo.duration.isDefined) {
-          totalExecutionTime - executorOverhead - gettingResultTime
+          math.max(totalExecutionTime - executorOverhead - gettingResultTime - schedulerDelay, 0)
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: I am not familiar with the Core module.
But is it possible that `totalExecutionTime - executorOverhead - gettingResultTime - schedulerDelay` is negative?
If so, is the calculation correct?

Comment (sddyljsx) [reply core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: It is impossible.
I write math.max(xxx, 0) here,  just be consistent with the calculation of the executorComputingTimeProportion 
L297:
 val executorComputingTimeProportion =
          math.max(100 - schedulerDelayProportion - shuffleReadTimeProportion -
            shuffleWriteTimeProportion - serializationTimeProportion -
            deserializationTimeProportion - gettingResultTimeProportion, 0)

Comment (sddyljsx) [reply core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: In fact,

executorComputingTime = executorRunTime - shuffleReadTime - shuffleWriteTime=totalExecutionTime - executorOverhead - gettingResultTime- schedulerDelay-shuffleReadTime - shuffleWriteTime

executorComputingTimeProportion = (totalExecutionTime - executorOverhead - gettingResultTime - schedulerDelay-shuffleReadTime - shuffleWriteTime) / totalExecutionTime * 100

but in the old code executorComputingTime does not minus schedulerDelay

**Comment by sddyljsx** [comment]
@srowen PTAL

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala (hunk 1 above)
Comment (srowen) [reply core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: I think it's fine for consistency and perhaps to account for rounding errors or something. I'm OK with it as the current calculation does seem obviously wrong in at least one case.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @sddyljsx , @lw-lin , @HyukjinKwon , @gengliangwang , @srowen .
Merged to master/2.4.

**Comment by sddyljsx** [comment]
Thanks for review! @srowen @dongjoon-hyun  @lw-lin @HyukjinKwon @gengliangwang

**Pull request closed without merging**